autoload/ale/fixers/precommit.vim

```diff
@@ -0,0 +1,27 @@
+" Author: Matthew Grossman <matt@mrgrossman.com>
+" Description: Fix multi-language pre-commit hooks
+
+
+call ale#Set('precommit_executable', 'pre-commit')
+call ale#Set('precommit_hooks', [])
+call ale#Set('precommit_options', '')
+
+function! ale#fixers#precommit#Fix(buffer) abort
```

Comment:
Another question @w0rp ; `pre-commit` doesn't spit the fixed file to stdout, but simply side-effects the files in place. When running the fixer locally, I essentially overwrite my own buffer with the output of stdout, which is something "useless" like

`Trim Trailing Whitespace.................................................Passed`

Is there a flag or return type I can modify the fix function to rely on the side-effecting nature of pre-commit to get this to work?

Reply:
```
(finished - exit code 0) ['/usr/local/bin/zsh', '-c', '''pre-commit'' run trailing-whitespace --files ''/var/folders/qz/j2l2wxfs18bbfh4vn6w00kwm0000gp/T/nvimf1gukG/13/abg_rental_provider.py'';']   
```

Reply:
The fixers are normally designed to read output from a command and then apply changes to the file, typically without writing the changes back to disk before you say so. What you could do for this is define a `process_with` callback which reloads the file and returns all of the lines in it with `return getbufline(a:buffer, 1, '$')`. See `:help ale-fix`.